gallery_dl/extractor/facebook.py

```diff
@@ -0,0 +1,454 @@
+
+        return photo
+
+    @staticmethod
+    def parse_post_page(post_page):
+        first_photo_url = text.extr(
+            text.extr(
+                post_page, '"__isMedia":"Photo"', '"target_group"'
+            ), '"url":"', ','
+        )
+
+        post = {
+            "set_id": text.extr(post_page, '{"mediaset_token":"', '"') or
+            text.extr(first_photo_url, 'set=', '"').rsplit("&", 1)[0]
+        }
+
+        return post
+
+    @staticmethod
+    def parse_video_page(video_page):
+        video = {
+            "id": text.extr(
+                video_page, '\\"video_id\\":\\"', '\\"'
+            ),
+            "username": FacebookExtractor.decode_all(text.extr(
+                video_page, '"actors":[{"__typename":"User","name":"', '","'
+            )),
+            "user_id": text.extr(
+                video_page, '"owner":{"__typename":"User","id":"', '"'
+            ),
+            "date": text.parse_timestamp(text.extr(
+                video_page, '\\"publish_time\\":', ','
+            )),
+            "type": "video"
+        }
+
+        if not video["username"]:
+            video["username"] = FacebookExtractor.decode_all(text.extr(
+                video_page,
+                '"__typename":"User","id":"' + video["user_id"] + '","name":"',
+                '","'
+            ))
+
+        first_video_raw = text.extr(
+            video_page, '"permalink_url"', '\\/Period>\\u003C\\/MPD>'
+        )
+
+        audio = {
+            **video,
+            "url": FacebookExtractor.decode_all(text.extr(
+                text.extr(
+                    first_video_raw,
+                    "AudioChannelConfiguration",
+                    "BaseURL>\\u003C"
+                ),
+                "BaseURL>", "\\u003C\\/"
+            )),
+            "type": "audio"
+        }
+
+        video["urls"] = {}
+
+        for raw_url in text.extract_iter(
+            first_video_raw, 'FBQualityLabel=\\"', '\\u003C\\/BaseURL>'
+        ):
+            resolution = raw_url.split('\\"', 1)[0]
+            video["urls"][resolution] = FacebookExtractor.decode_all(
+                raw_url.split('BaseURL>', 1)[1]
+            )
+
+        if not video["urls"]:
+            return video, audio
+
+        video["url"] = sorted(
+            video["urls"].items(),
+            key=lambda x: int(x[0][:-1])
+        )[-1][1]
+
+        video["filename"] = text.rextract(video["url"], "/", "?")[0]
+        FacebookExtractor.item_filename_handle(video)
+
+        audio["filename"] = video["name"] + ".m4a"
+        FacebookExtractor.item_filename_handle(audio)
+
+        return video, audio
+
+    def photo_page_request_wrapper(self, url, *args, **kwargs):
+        LEFT_OFF_TXT = "" if url.endswith("&set=") else (
+            "\nYou can use this URL to continue from "
+            "where you left off (added \"&setextract\"): "
+            "\n" + url + "&setextract"
+        )
+
+        res = self.request(url, *args, **kwargs)
```

Comment:
```suggestion
        res = self.request(url, **kwargs)
```

`Extractor.request()` has only `url` as positional argument.